# Arithmetic and Variables

## Printing

Printing dalam bahasa pemrograman Python mengacu pada proses output atau menampilkan hasil dari program ke layar atau ke media cetak. Untuk mencetak atau menampilkan nilai atau teks ke layar dalam Python, Anda dapat menggunakan fungsi print(). Fungsi print() dapat menerima satu atau lebih argumen dan mencetaknya ke layar dalam bentuk teks. 

Contoh:

In [2]:
print("Hello, World!")

Hello, World!


Hasil dari kode di atas adalah mencetak teks "Hello, World!" ke layar. 

Berikut contoh-contoh lain menggunakan fungsi print() dalam Python:

In [3]:
print("Secret World")

Secret World


In [14]:
print = ("Secret World")
print

'Secret World'

In [7]:
nama = "Alice"
umur = 30
print("Nama:", nama)
print("Umur:", umur)

Nama: Alice
Umur: 30


## Arithmetic

Arithmetic dalam bahasa pemrograman Python mengacu pada operasi matematika dasar seperti penjumlahan, pengurangan, perkalian, dan pembagian. Python mendukung semua operasi aritmatika dasar ini menggunakan operator khusus seperti + (penjumlahan), - (pengurangan), * (perkalian), dan / (pembagian). 

Contoh:

In [2]:
print(10 + 20)

30


Maka hasil dari perkalian di atas adalah 30. 

Berikut contoh-contoh lain menggunakan dari Arithmetic pada Python:

In [3]:
print(30 * 2 * 5 - 100 + 60)

260


In [4]:
print(200 / 4 * 7 - 10 + 150)

490.0


## Variable

Variables dalam Python adalah tempat untuk menyimpan nilai. Variabel memungkinkan Anda menyimpan data dalam program Anda sehingga dapat diakses dan digunakan kembali nanti. Untuk membuat variabel, Anda cukup menetapkan nama variabel tersebut dan nilai yang ingin Anda simpan ke dalamnya menggunakan tanda sama dengan (=). 

Contoh:

In [1]:
x = 5
y = 10
print(x * y)

50


Mari saya jelaskan langkah-langkahnya:

1. Anda mendefinisikan dua variabel `x` dan `y` dengan nilai masing-masing 5 dan 10.
2. Anda menggunakan fungsi `print()` untuk mencetak nilai dari variabel `X`dan `y` ke layar.

Hasilnya akan mencetak nilai 50, karena hasil perkalian dari `x` dan `y`, yang masing-masing bernilai 5 dan 10. Sehingga 5 * 10 = 50.




Berikut contoh-contoh lain menggunakan dari Variable pada Python:

In [11]:
a = 50 
b = 100
c = 1

print(a, b, c)

50 100 1


In [12]:
k = 43
l = 57
m = 109
n = k * m - l

print(n)

4630


## Comments

Dalam Python, komentar digunakan untuk memberikan penjelasan atau dokumentasi pada kode. Komentar tidak dieksekusi oleh interpreter Python dan hanya digunakan sebagai panduan untuk pengembang atau orang lain yang membaca kode.

Contoh:

In [13]:
# Ini adalah baris comments

Dalam Python, komentar satu baris dimulai dengan tanda pagar (#). Komentar jenis ini digunakan untuk memberikan penjelasan atau dokumentasi singkat mengenai baris kode yang bersangkutan. Komentar tersebut tidak akan dieksekusi oleh interpreter Python dan hanya berfungsi sebagai panduan atau catatan bagi pembaca kode.

Berikut contoh-contoh lain penggunaan Comments pada Python:

In [18]:
# Ini adalah contoh komentar multi-baris dalam Python.
# Komentar ini dapat mencakup beberapa baris kode.

In [16]:
x = 5  # inisialisasi variabel x dengan nilai 5

### Creating Variable

Kode selanjutnya membuat sebuah variabel.

Contoh:

In [21]:
# creat variable called multiply_test and give it a value of 15*7
multiply_test = 15 * 7

# print the value of multiply_test
print(multiply_test)

105


Kode tersebut terdiri dari dua baris. 

1. Pada baris pertama, kita membuat sebuah variabel yang dinamai `multiply_test` dan memberikannya nilai dengan melakukan operasi perkalian antara 15   dan 7 (`15 * 7`). Hasil dari operasi tersebut, yaitu 105, akan disimpan di dalam variabel `multiply_test`.
   
2. Pada baris kedua, kita mencetak nilai yang disimpan di dalam variabel `multiply_test` dengan menggunakan fungsi `print()`. Ini akan menampilkan nilai 105 di layar. 

Jadi, dengan menjalankan kedua baris kode ini, kita akan mendapatkan output yang menampilkan nilai 105.


Berikut contoh-contoh lain penggunaan Create Variable pada Python:

In [25]:
# create variable called subtraction_test and give it a value 100-90
subtraction_test = 100 - 90

# print the value of subtraction_test
print(subtraction_test)

10


In [26]:
# buatlah angka yang bernama tes_angka dan berikan nilai padanya 1000*4/50
tes_angka = 1000 * 4 / 50

# cetaklah nilai dari tes_angka
print(tes_angka)

80.0


### Manipulating Variable

Manipulasi variabel dalam Python merujuk pada proses mengubah nilai variabel atau melakukan operasi matematika, logika, atau manipulasi data lainnya menggunakan variabel sebagai wadah atau penyimpanan.

Contoh:

In [29]:
# set the value of a new variable to 59
my_var = 59

# print the value assigned to my_var
print(my_var)

# change the new value of the variable to 10
my_var = 10

# print the new value assigned to my_var
print(my_var)

59
10


Pada kode di atas, terdapat serangkaian operasi yang dilakukan terhadap variabel `my_var`. 

1. Pertama, sebuah variabel baru `my_var` diberi nilai 59 dengan pernyataan `my_var = 59`.
2. Nilai yang ditetapkan ke `my_var` dicetak menggunakan pernyataan `print(my_var)`, sehingga outputnya adalah 59.
3. Kemudian, nilai variabel `my_var` diubah menjadi 10 dengan pernyataan `my_var = 10`.
4. Nilai baru yang ditetapkan ke `my_var` dicetak lagi dengan menggunakan pernyataan `print(my_var)`, sehingga outputnya berubah menjadi 10.

Jadi, secara singkat, kode tersebut menunjukkan bagaimana nilai variabel `my_var` bisa diatur, dicetak, dan kemudian diubah nilainya.

Berikut contoh-contoh lain penggunaan Manipulating Variable pada Python:

In [30]:
# atur nilai dari sebuah variabel baru menjadi 99
angka_baru = 99

# cetak nilai yang ditetapkan untuk angka_baru
print(angka_baru)

# ubah variabel baru menjadi 39
angka_baru = 39

# cetak nilai baru yang ditetapkan untuk angka_baru
print(angka_baru)

99
39


In [32]:
# atur nilai s_s menjadi 19
s_s = 19

# cetak nilai s_s
print(s_s)

# ubah nilai s_s menjadi 45
s_s = 45

# cetak nilai s_s yang baru
print(s_s)

19
45


In [33]:
# cetak angka_baru dan s_s
print(angka_baru)
print(s_s)

39
45


In [34]:
# kalikan angka-baru dengan 5
angka_baru = angka_baru * 5

# cetaklah angka_baru
print(angka_baru)

195


### Using Multiple Variables

Seringkali dalam kode digunakan beberapa variabel. Ini sangat berguna terutama ketika kita harus melakukan perhitungan panjang dengan beberapa masukan.

Contoh:

In [38]:
# buat variabel
angka_tahun = 12
hari_per_tahun = 365
jam_per_hari = 24
menit_per_jam = 60
detik_per_menit = 60

# jumlah detik dari 12 tahun
total_detik = detik_per_menit * menit_per_jam * jam_per_hari * hari_per_tahun * angka_tahun

print(total_detik)

378432000


Kode di atas bertujuan untuk menghitung jumlah detik dari 12 tahun. Prosesnya adalah sebagai berikut:

1. Variabel `angka_tahun` diberi nilai 12.
2. Variabel `hari_per_tahun` diberi nilai 365.
3. Variabel `jam_per_hari` diberi nilai 24.
4. Variabel `menit_per_jam` diberi nilai 60.
5. Variabel `detik_per_menit` diberi nilai 60.
6. Kemudian, total detik dihitung dengan mengalikan semua komponen waktu, yaitu detik per menit, menit per jam, jam per hari, hari per tahun, dan angka tahun.
7. Hasilnya dicetak menggunakan pernyataan `print()`.

Jadi, kode tersebut menghasilkan jumlah detik dari 12 tahun dengan mengalikan semua konversi waktu yang diberikan.


Perlu dicatat bahwa penggunaan variabel sangat berguna ketika nilai dari masukan dapat berubah. Misalnya, katakanlah kita ingin sedikit meningkatkan perkiraan kita dengan memperbarui nilai jumlah hari dalam setahun dari 365 menjadi 365,25, untuk memperhitungkan tahun kabisat. Kemudian kita dapat mengubah nilai yang ditetapkan ke `hari_per_tahun` tanpa mengubah variabel lainnya dan melakukan perhitungan ulang.

Contoh:

In [41]:
# pembaruan untuk menyertakan tahun kabisat
hari_per_tahun = 365.25

# jumlah detik dari 12 tahun
total_detik = detik_per_menit * menit_per_jam * jam_per_hari * hari_per_tahun * angka_tahun

print(total_detik)

378691200.0


### Debugging

Salah satu kesalahan umum saat bekerja dengan variabel adalah secara tidak sengaja membuat kesalahan ketik.

Contoh:

In [45]:
print(total_dtk)

NameError: name 'total_dtk' is not defined

Ketika Anda melihat NameError seperti ini, itu menunjukkan bahwa Anda harus memeriksa bagaimana Anda mengeja variabel yang dirujuknya sebagai "tidak didefinisikan". Kemudian, untuk memperbaiki kesalahan, Anda hanya perlu memperbaiki pengejaan.

Contoh:

In [46]:
print(total_detik)

378691200.0
